### ISSUE TITLE
Observations as of 0.13

### ISSUE THREAD

**Issue by keness521** [ebaauw/homebridge-deconz#13]
I love the leaner (since I don't use Hue) Homebridge-deCONZ plugin! And it is becoming completely usable for daily use much more rapidly that I thought, so thank you very much!

I have noticed a couple of things, that I thought I might post about.

1) EXPOSE LIGHTS - I'm not sure if I am understanding this correctly, but I thought if I left the main EXPOSE 'ON' and turned EXPOSE LIGHTS to 'off' the light devices would either disappear or become No Response in HomeKit. When I did this, they never went away. The lights continued to show up in Home app and be function even after waiting 30 minutes. When I turned the main EXPOSE to 'off' everything disappeared, and I had to re-name, re-room re-scene all devices when I turned the main EXPOSE back 'on'.

2) Adding devices - One of the nice benefits of the new plugin is not needing to restart Homebridge when you add devices! I've noticed a bit of a drawback, too, though... When I add devices in Phoscon, I always name them of course, and in the past, that naming carried over to HomeKit because by the time they were exposed to HomeKit, the renaming was complete. The new plugin exposes them SO fast that I haven't yet had a chance to rename them, so HomeKit gets the generic name. I added a bunch of temperature sensors at one time, renaming them in Phoscon, like I used to do, but then when I went to Home app, I had 10 new temperature sensors, which I would need to rename again, but worse, I didn't know which one was which, so I had to delete everything and go back and re-add them 1-by-1, renaming them twice, once in Phoscon, then again in Home app. Is there a better way, am I doing something wrong?

**Comment by ebaauw** [comment]
Thanks for your feedback.  I'm still trying to find the best way of providing the settings to expose devices.

Currently, _Expose Lights_, _Expose Sensors_, and _Expose Groups_ are more like auto-expose settings.  When set, all corresponding devices are exposed, with a _Device Settings_ service to un-expose each accessory individually.  When an accessory is un-exposed, the _Device Settings_ service "moves" to the Gateway accessory, so you can re-expose it.  When un-selecting _Expose Lights_, etc, the accessories continue to be exposed, but the _Device Settings_ services on the Gateway accessory are removed.  Also, new devices won't be exposed automatically.  When later re-selecting _Expose Lights_, ..., the previously blacklisted devices remain blacklisted, but the _Device Settings_ service re-appears under the Gateway accessory, and accessories for new devices will be added.

The _Expose_ setting on the gateway is like a hard reset: de-selecting it clears all configuration settings for that gateway, incl. the blacklist, and even the API key.  When re-selecting _Expose_, the gateway is treated like a new gateway, and a new API key needs to be obtained.

By un-exposing (deselecting _Expose_ on the _Device Settings_ service of the device accessory) and re-exposing (selecting _Expose_ on the _Device Settings_ service in the gateway accessory), the accessory is reset.  That should refresh the names.  No need to remove them from deCONZ.  This also clears the Eve history for the accessory.

As I said: I'm still struggling with the best way to deal with this.  I thought about exposing all devices by default, but on large installations, you'd run into the 149 accessories per bridge limit.  I thought about not exposing any device by default, but you'd run into the 99 services per accessory limit.  I thought about exposing the _Device Settings_ on the gateway accessory, regardless of whether the device is blacklisted, but again: the 99 services per accessory limit.  Maybe I could provide a way to scroll through the devices.  In that case, I'd have a single _Configure_ characteristic to expose the _Device Settings_ services on the gateway, like for 50 devices at a time.  
I also like to be able to change how to expose a device (e.g. from _Lightbulb_ to _Outlet_), without removing the accessory from HomeKit.  Of course, you'd lose the scenes and automations (as the characteristics now belong to a different service), but you'd keep the room assignment.

I'm also exploring the possibility to do these settings from the Homebridge UI.  The challenge is that it runs detached from Homebridge, so I need a mechanism to apply the changed settings without having to restart Homebridge.  Of course, that would mean that the plugin is usable only from a standard Homebridge installation, with the UI.  Then again, now you need Eve, or another HomeKit app, and to be the owner of the HomeKit home, to be able to do the configuration.

**Comment by keness521** [comment]
I think I understand better how the various EXPOSE options work now, or at least can refer back to this message to refresh!

I think a configuration method through Homebridge sounds tempting. My thinking is:

 - We already have to have Homebridge, so it isn't an additional app (like Eve) that we might not already be using. I'm sure some people are using it anyway, but certainly not all.
 - No 'Owner of the Home' requirement.
 - Homebridge makes it so easy to put any plugin into a Child Bridge that even if you did have to restart Homebridge, you'd only be restarting the Child Bridge not the entire Homebridge. (I suspect that psychologically this is a very undesirable option for you, since one of the biggest goals of the new plugin was to be dynamic and not require a restart. Although it would solve the problem of accessories popping into HomeKit before they'd been fully renamed in Phoscon.)

You mentioned that doing configuration through the Homebridge UI requires a standard Homebridge installation. I wonder how many non-standard installations would be likely for this plugin that would conflict with that. Or if a separate web-based configuration page on its own port, outside of Homebridge UI, might be an option that would work with any type of Homebridge installation. Plus, it might be a way to quickly and easily rename newly added devices (Phoscon is cumbersome on mobile) and an easily clickable "refresh" button could then update the changed name or other setting into HomeKit without needing to go anywhere else to make it happen, and while retaining dynamic loading.

**Comment by ebaauw** [comment]
I started doing Homebridge plugins, so I wouldn’t have to deal with user interfaces.  Creating a separate web-based UI is way too much for me.

Your expose before renaming issue would also be solved by not automatically exposing new devices, but requiring a manual action.

Names of accessories and services are accessible only to HomeKit apps; it’s technically not possible for a Homebridge plugin (or any accessory) to change these.  It can only un-expose and re-expose the accessory, so HomeKit re-initialises the names, but loses the configuration.

I don’t collect any statistics, but judging by the issues I get on GitHub, and the questions on Discord, there are more non-standard installations out there than installations without a proper HomeKit app.

BTW, historically, Apple’s Home is the additional app.  HomeKit started out as a framework, where each accessory vendor would deliver their own app.  When I started Homebridge Hue, Eve and Home+ were already there.  Apple’s Home app was introduced later, and it still doesn’t support all of HomeKit.
